make fails

# Issue type
Installation

# Version found
  * git hash of the version you're running `git rev-parse HEAD`
6eb2f4b810fc62058267ca18a478ada020353b8e

# Your environment and setup
  * Operating System name and version (usually `uname -a`)
Linux q552-0503 4.4.0-31-generic #50~14.04.1-Ubuntu SMP Wed Jul 13 01:07:32 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
  * Are you running from the command line? From a container?
From the command line.
  * Did you pass in special parameters when building the app?
$ CXX=g++-5 ./configure

# Issue description
I have failed to install stellar-core on Ubuntu 14.04 according to Installation Instructions.

## Actual result
~/stellar-core$ make
...
ledger/LedgerStateTests.cpp:1891:17: error: converting to ‘const std::tuple<stellar::Asset, stellar::Asset, long int>’ from initializer list would use explicit constructor ‘constexpr std::tuple< <template-parameter-1-1> >::tuple(_UElements&& ...) [with _UElements = {stellar::Asset&, stellar::Asset&, int}; <template-parameter-2-2> = void; _Elements = {stellar::Asset, stellar::Asset, long int}]’
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-unknown-warning-option’
cc1plus: warning: unrecognized command line option ‘-Wno-unused-local-typedef’
cc1plus: warning: unrecognized command line option ‘-Wno-unused-command-line-argument’
make[3]: *** [ledger/LedgerStateTests.o] Error 1
make[3]: Leaving directory `/home/deploy/stellar-core/src'
make[2]: *** [all] Error 2
make[2]: Leaving directory `/home/deploy/stellar-core/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/deploy/stellar-core'
make: *** [all] Error 2

## Expected result
~/stellar-core$ make
...
make: *** [all] Error 0

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -21,7 +21,7 @@ When running a node, the best bet is to go with the latest release.
 
 ## Build Dependencies
 
-- `clang` >= 5.0 or `g++` >= 5.0
+- `clang` >= 5.0 or `g++` >= 6.0
 - `pkg-config`
 - `bison` and `flex`
 - `libpq-dev` unless you `./configure --disable-postgres` in the build step below.

--- a/configure.ac
+++ b/configure.ac
@@ -46,6 +46,14 @@ AX_FRESH_COMPILER
 # -pthread seems to be required by -std=c++14 on some hosts
 AX_APPEND_COMPILE_FLAGS([-pthread])
 
+AC_MSG_CHECKING([whether defect report N4387 is resolved])
+AC_COMPILE_IFELSE([AC_LANG_SOURCE([[#include <tuple>
+std::tuple<int, int> f()
+{
+    return {1, 2};
+}
+]])], [AC_MSG_RESULT([yes])], [AC_MSG_RESULT([no]); AC_MSG_ERROR([defect report N4387 is not resolved])], AC_MSG_FAILURE)
+
 AC_MSG_CHECKING([for c++14 compliant std::weak_ptr move-constructor])
 AC_RUN_IFELSE([AC_LANG_PROGRAM([[#include <memory>]], [[std::shared_ptr<int> shared = std::make_shared<int>(0);
 std::weak_ptr<int> weak1(shared);

